### PR TITLE
fix: workers.New err must be returned

### PR DESCRIPTION
**Pull request by jiuker**
## Description

When workers.New get err should return

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

File: cmd/bucket-lifecycle.go [hunk 1]
```diff
@@ -157,11 +157,13 @@ func initBackgroundExpiry(ctx context.Context, objectAPI ObjectLayer) {
 	ewk, err := workers.New(workerSize)
 	if err != nil {
 		logger.LogIf(ctx, err)
+		return
```
Comment (harshavardhana) [inline cmd/bucket-lifecycle.go]: actually this PR is wrong we should always initialize ILM and never return errors, initialize this with some random valid value.

```suggestion
		ewk = 4
```

File: cmd/bucket-lifecycle.go [hunk 2]
```diff
@@ -157,11 +157,13 @@ func initBackgroundExpiry(ctx context.Context, objectAPI ObjectLayer) {
...
 	}
 
 	nwk, err := workers.New(workerSize)
 	if err != nil {
 		logger.LogIf(ctx, err)
+		return
```
Comment (harshavardhana) [inline cmd/bucket-lifecycle.go]:
```suggestion
		nwk = 4
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by jiuker** [comment]
@harshavardhana it seems that is wrong which you suggested.Do you mean workerSize should be 4 by default?

**Comment by harshavardhana** [comment]
> @harshavardhana it seems that is wrong which you suggested.Do you mean workerSize should be 4 by default?

Correct if someone sets the wrong value @jiuker - we should never fail to initialize, we should always initialize with sane defaults.

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]

**Pull request merged**